conda/common/configuration.py

```diff
@@ -1609,3 +1609,69 @@ def typify_parameter(self, parameter_name, value, source):
 
     def get_descriptions(self):
         raise NotImplementedError()
+
+
+def get_plugin_config_data(
+    data: dict[Path, dict[str, RawParameter]],
+) -> dict[Path, dict[str, RawParameter]]:
+    """
+    This is used to move everything under the key "plugins" from the provided dictionary
+    to the top level of the returned dictionary. The returned dictionary is then passed
+    to :class:`PluginConfig`.
+    """
+    new_data = {}
+
+    for source, config in data.items():
+        if plugin_data := config.get("plugins"):
+            for param_name, raw_param in plugin_data.value(None).items():
+                new_data[source] = {param_name: raw_param}
+
+        if source == "envvars":
```

Comment:
I don't think so because there are no error conditions in this function. Here, we simply walk through the dictionary that's provided to try and pluck out what we want for the return dictionary. If don't find anything, we simply return an empty dictionary.